lib/cli/scripts/licenses.ts

```diff
@@ -148,29 +151,34 @@ export default function main(_args: string[], workingDir: string) {
                     console.error(err);
                     reject(err);
                 } else {
+                    const filteredPackages = {};
+
                     // eslint-disable-next-line guard-for-in
                     for (const packageName in packages) {
                         const pack = packages[packageName];
-                        pack['licenseExp'] = pack['licenses']
-                            .toString()
-                            .replace(/\*/g, '')
-                            .replace(/[a-zA-Z0-9\-.]+/g, (match: string) => {
-                                const lowerMatch = match.toLowerCase();
-
-                                if (lowerMatch !== 'and' && lowerMatch !== 'or' && lowerMatch !== 'with') {
-                                    return licenseWithMDLinks(match);
-                                } else {
-                                    return match;
-                                }
-                            });
-
-                        if (!pack['repository']) {
-                            const lastAtSignPos = packageName.lastIndexOf('@');
-                            const mainName = packageName.substring(0, lastAtSignPos);
-
-                            if (missingRepositories[mainName]) {
-                                pack['repository'] = missingRepositories[mainName];
-                            }
+
+                        const lastAtSignPos = packageName.lastIndexOf('@');
+                        const basePackageName = packageName.substring(0, lastAtSignPos);
+
+                        if (mainDependencies.includes(basePackageName)) {
+                            pack['licenseExp'] = pack['licenses']
+                                .toString()
+                                .replace(/\*/g, '')
+                                .replace(/[a-zA-Z0-9\-.]+/g, (match) => {
```

Comment:
seems like type of match is any, could you please set it to string as it seems to be string?
<img width="643" alt="Screenshot 2024-10-16 at 12 04 46" src="https://github.com/user-attachments/assets/eb0274a7-cf15-4f91-9b45-cd9aaeed1271">

Reply:
added